### DISCUSSION TITLE
How to debug what's causing `fwupdmgr security` to emit a 'not enough info' result?

### DISCUSSION THREAD

**Discussion by KhalilSantana** [fwupd/fwupd#6050] [Q&A]
Hi,

It seems fwupdmgr (1.9.3 on Arch) is not finding some information it needs in order to provide an HSI report on my system, providing the only the following information:

```
Not enough data was provided by the platform to make an HSI calculation
 » https://fwupd.github.io/hsi.html#not-enough-info
khalil:~ % Not enough data was provided by the plat
```

However, how could I derive _what_ information fwupdmgr needs to do that? Is there a list of prerequisites for this that perhaps I'm not following? In short, a more complete error message like "couldn't read tpm2.0 firmware version" or "missing <foo> service to query <data> would be very helpful.

Thus far I've tried stracing it to see what it's attempting (and failing) to do, but it's rather low level so all I've got is a bunch of

```
read(3, 0x7ffed71b83a0, 8)              = -1 EAGAIN
```
Before the error is emitted, so it's not trivial to see _what_ it was attempting to read without attaching a debugger and such. Any tips?

**Comment by superm1** [comment]
The kind of stuff it looks for isn't _usually_ things that you can manually influence.  Is this an AMD system?  If so; do you have the `ccp` kernel module enabled?  If so, can you look at in `ls /sys/bus/pci/drivers/ccp/*`?

**Comment by KhalilSantana** [comment]
>  Is this an AMD system?

Yes, AsRock B650 PG Riptide + 7800X3D

> > If so; do you have the ccp kernel module enabled?

It seems so:

``` 
% lsmod | grep ccp
ccp                   155648  1 kvm_amd
```
> If so, can you look at in ls /sys/bus/pci/drivers/ccp/*?

```
# ls /sys/bus/pci/drivers/ccp/*
/sys/bus/pci/drivers/ccp/bind  /sys/bus/pci/drivers/ccp/new_id  /sys/bus/pci/drivers/ccp/remove_id  /sys/bus/pci/drivers/ccp/uevent  /sys/bus/pci/drivers/ccp/unbind

'/sys/bus/pci/drivers/ccp/0000:0f:00.2':
ari_enabled           config                    current_link_width  dma_mask_bits    enable         iommu_group  local_cpulist   max_link_width  msi_irqs   power_state  resource   revision          subsystem_vendor
broken_parity_status  consistent_dma_mask_bits  d3cold_allowed      driver           firmware_node  irq          local_cpus      modalias        numa_node  remove       resource2  subsystem         uevent
class                 current_link_speed        device              driver_override  iommu          link         max_link_speed  msi_bus         power      rescan       resource5  subsystem_device  vendor

/sys/bus/pci/drivers/ccp/module:
coresize  drivers  holders  initsize  initstate  notes  parameters  refcnt  sections  srcversion  taint  uevent  version
```

**Reply by superm1** [reply]
Assuming this is a relatively modern kernel; I don't see any of the sysfs files necessary for doing a calculation on an AMD system, which means your platform doesn't export the feature.
[PSP_CAPABILITY_PSP_SECURITY_REPORTING](https://github.com/torvalds/linux/blob/5d0c230f1de8c7515b6567d9afba1f196fb4e2f4/drivers/crypto/ccp/sp-pci.c#L80)

**Comment by KhalilSantana** [comment]
After applying the BIOS update (1.24 → 1.28) fwupdmgr can gather HSI information from my system, so it seems ASRock / AMD fixed it. Thank you for your assistance!
